lib/dialects/abstract/query-generator.js

```diff
@@ -1520,6 +1522,195 @@ module.exports = (function() {
       return result;
     },
 
+    whereQuery: function(where, options) {
+      var query = this.whereItemsQuery(where, options);
+      if (query && query.length) {
+        return 'WHERE '+query;
+      }
+      return '';
+    },
+    whereItemsQuery: function(where, options, binding) {
+      if (
+        (Array.isArray(where) && where.length === 0) ||
+        (_.isPlainObject(where) && _.isEmpty(where)) ||
+        where === '' ||
+        where === null ||
+        where === undefined
+      ) {
+        // NO OP
+        return '';
+      }
+
+      var self = this
+        , items = [];
+
+      binding = ' '+ (binding || 'AND') + ' ';
+
+      if (_.isPlainObject(where)) {
+        _.forIn(where, function (value, key) {
```

Comment:
This should probably be forOwn, to avoid problems where users modify Object.prototype